nodejs/awsx/apigateway/README.md

```diff
@@ -98,6 +98,121 @@ A complete example can be found [here](https://github.com/pulumi/pulumi-awsx/blo
+                console.log("Received event:", JSON.stringify(event, null, 2));
+                return {
+                    principalId: "user",
+                    policyDocument: {
+                        Version: "2012-10-17",
+                        Statement: [{
+                            Action: "execute-api:Invoke",
+                            Effect: "Allow",
+                            Resource: event.methodArn,
+                        }],
+                    },
+                };
+            },
+            identitySource: ["method.request.querystring.auth"],
+        }],
+    }],
+});
+```
+
+Below is an example of a custom "token" Lambda Authorizer.
+
+```ts
+import * as awsx from "@pulumi/awsx";
+
+const api = new awsx.apigateway.API("myapi", {
+    routes: [{
+        path: "/b",
+        method: "GET",
+        eventHandler: async () => {
+            return {
+                statusCode: 200,
+                body: "<h1>Hello world!</h1>",
+            };
+        },
+        authorizers: [{
+            parameterName: "Authorization",
+            parameterLocation: "header",
+            authType: "oauth2",
+            type: "request",
+            handler: async (event: awsx.apigateway.AuthorizerEvent) => {
+                // Add your own custom authorization logic here.
+                // Access the token using event.authorizationToken
+                console.log("Received event:", JSON.stringify(event, null, 2));
+                return {
+                    principalId: "user",
+                    policyDocument: {
+                        Version: "2012-10-17",
+                        Statement: [{
+                            Action: "execute-api:Invoke",
+                            Effect: "Allow",
+                            Resource: event.methodArn,
+                        }],
+                    },
+                };
+            },
+            identitySource: ["method.request.querystring.auth"],
+        }],
+    }],
+});
+```
+
+You can also define the Lambda Authorizer elsewhere and then reference the required values.
+
+```ts
+import * as awsx from "@pulumi/awsx";
+
+const apiWithAuthorizer = new awsx.apigateway.API("authorizer-api", {
+    routes: [{
+        ...
+        authorizers: [{
+            authorizerName: "testing",
+            parameterName: "auth",
+            parameterLocation: "query",
+            authType: "custom",
+            type: "request",
+            handler: {
+                uri: pulumi.interpolate`arn:aws:apigateway:${region}:lambda:path/2015-03-31/functions/${authorizerLambda.arn}/invocations`,
+                credentials: gatewayRole.arn,
+            },
+            identitySource: ["method.request.querystring.auth"],
+        }],
+    }],
+});
+```
+
+A complete example of defining the Lambda Authorizer elsewhere can be found [here](https://github.com/pulumi/pulumi-awsx/blob/master/nodejs/awsx/examples/api/index.ts).
+
```

Comment:
in general, use a permalink in github so that we always reference something at a known good point in time.  that way, for example, if we remove the code from this example and move it elsewhere, the permalink is still valid.